### ISSUE TITLE
Consider making the execution of the Cache post-job step configurable

### ISSUE THREAD

**Issue by MatthewSteeples** [microsoft/azure-pipelines-tasks#13194]
## Required Information

Entering this information will route you directly to the right team and expedite traction.

**Question, Bug, or Feature?**  
*Type*: Feature

**Enter Task Name**: Cache 

## Environment
- Server - Azure Pipelines
Account: ledgerscope
Project: Core
Build: CI
Build Id: 23425
    
- Agent - Hosted 
    
    - If using Hosted agent, provide agent queue name: Azure Pipelines

## Issue Description

We have a single CI build which runs 2 different Jobs in parallel for each commit. We're using caching to reduce the amount of time we spend downloading dependencies. When the cache key changes, there is frequently a race condition when updating the cache at the end of the Job because the builds take a near-identical amount of time to run. They both check for the presence of the key, but as it takes 30 seconds to update the cache, they end up both uploading it (or at least doing the work for it).

This doesn't cause a problem (as far as I'm aware), it just means that the Job is doing more work than it needs to.

Would it be possible to make the Cache Task configurable so that (for example) we could ensure that only one of the Jobs is responsible for updating the cache if there was a cache miss?

**Comment by StephenHodgson** [comment]
Bunp

**Comment by StephenHodgson** [comment]
Bippity bumpitty boop

**Comment by emibcn** [comment]
Bump?

**Comment by StephenHodgson** [comment]
Agreed bump

**Comment by emibcn** [comment]
Bump?

**Comment by StephenHodgson** [comment]
I think you gotta open a new ticket. Stale-Bot is unforgiving and msft seems to be unwilling to make this change.

**Comment by emibcn** [comment]
Ok, never mind. Our team at ABB will migrate out of MSFT.
